contracts/test/StakeTableV2Governance.invariant.t.sol

```diff
@@ -0,0 +1,537 @@
+    function transferOwnership(address newOwner) public {
+        // Bound newOwner to valid addresses (not zero)
+        newOwner = _boundAddress(newOwner);
+
+        vm.startPrank(currentAdmin);
+        proxy.transferOwnership(newOwner);
+        currentAdmin = newOwner;
+        _addActor(newOwner);
+        transferOwnershipCalls++;
+        vm.stopPrank();
+    }
+
+    /// @notice Fuzzed action: Grant DEFAULT_ADMIN_ROLE
+    function grantAdminRole(address newAdmin) public {
+        newAdmin = _boundAddress(newAdmin);
+
+        vm.startPrank(currentAdmin);
+        proxy.grantRole(proxy.DEFAULT_ADMIN_ROLE(), newAdmin);
+        currentAdmin = newAdmin;
+        _addActor(newAdmin);
+        grantRoleCalls++;
+        vm.stopPrank();
+    }
+
+    /// @notice Fuzzed action: Grant PAUSER_ROLE
+    function grantPauserRole(address newPauser) public {
+        newPauser = _boundAddress(newPauser);
+
+        vm.startPrank(currentAdmin);
+        proxy.grantRole(proxy.PAUSER_ROLE(), newPauser);
+        _addActor(newPauser);
+        grantRoleCalls++;
+        vm.stopPrank();
+    }
+
+    /// @notice Fuzzed action: Revoke PAUSER_ROLE
+    /// @dev Should always succeed when called by admin (even if actor doesn't have role)
+    function revokePauserRole(uint256 actorIndexSeed) public {
+        if (actors.length == 0) return;
+
+        address actor = actors[actorIndexSeed % actors.length];
+
+        vm.startPrank(currentAdmin);
+        proxy.revokeRole(proxy.PAUSER_ROLE(), actor);
+        revokeRoleCalls++;
+        vm.stopPrank();
+    }
+
+    /// @notice Fuzzed action: Attempt unauthorized transfer
+    /// @dev Should always fail - verifies access control
+    function unauthorizedTransferOwnership(uint256 actorIndexSeed, address newOwner) public {
+        if (actors.length == 0) return;
+
+        address actor = actors[actorIndexSeed % actors.length];
+        newOwner = _boundAddress(newOwner);
+
+        // Only try if actor is NOT the current admin
+        if (actor == currentAdmin) return;
+
+        vm.startPrank(actor);
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IAccessControl.AccessControlUnauthorizedAccount.selector,
+                actor,
+                proxy.DEFAULT_ADMIN_ROLE()
+            )
+        );
+        proxy.transferOwnership(newOwner);
+        vm.stopPrank();
+    }
+
+    /// @notice Fuzzed action: Transfer ownership to self (tests idempotency)
+    /// @dev Should always succeed - tests that self-transfer doesn't break state
+    function transferOwnershipToSelf() public {
+        vm.startPrank(currentAdmin);
+        proxy.transferOwnership(currentAdmin);
+        transferOwnershipCalls++;
+        vm.stopPrank();
+    }
+
+    /// @notice Fuzzed action: Chain of transfers
+    /// @dev Should always succeed - tests multiple sequential transfers
+    function chainOfTransfers(address[] memory newOwners) public {
```

Comment:
I think `transferOwnership` already handles this. It will be called right after each other sometimes when foundry is running the test sequence.